testing/cloudbuild-templates/cloud-run-anthos-template.cloudbuild.yaml

```diff
@@ -0,0 +1,97 @@
+steps:
+- id: 'Generate UUID'
+  name: 'ubuntu'
+  entrypoint: '/bin/bash'
+  args:
+  - '-c'
+  - |
+    cat /dev/urandom | env LC_CTYPE=C tr -dc 'a-z0-9' | fold -w 15 | head -n 1 > _uuid
+
+- id: 'Build Container Image'
+  name: 'gcr.io/cloud-builders/docker:latest'
+  entrypoint: /bin/bash
+  dir: '${_SAMPLE_DIR}'
+  args:
+  - -c
+  - |
+    docker build -t gcr.io/${PROJECT_ID}/${_SERVICE}:$(/bin/cat _uuid) .
+
+- id: 'Push Container Image'
+  name: 'gcr.io/cloud-builders/docker:latest'
+  entrypoint: /bin/bash
+  args:
+  - -c
+  - |
+    docker push gcr.io/${PROJECT_ID}/${_SERVICE}:$(/bin/cat _uuid)
+
+- id: 'Deploy to Cloud Run'
+  name: 'gcr.io/cloud-builders/gcloud:$_CLOUDSDK_VERSION'
+  entrypoint: /bin/bash
+  args:
+  - -c
+  - |
+    gcloud run deploy ${_SERVICE}-$(/bin/cat _uuid) \
+     --image gcr.io/${PROJECT_ID}/${_SERVICE}:$(/bin/cat _uuid) \
+     --cluster ${_CLUSTER} \
+     --cluster-location ${_CLUSTER_LOCATION} \
+     --platform gke
+
+- id: 'Get Cloud Run URL'
+  name: 'gcr.io/cloud-builders/gcloud:$_CLOUDSDK_VERSION'
+  entrypoint: '/bin/bash'
+  args:
+  - '-c'
+  - |
+    get_url() {
+        gcloud run services describe ${_SERVICE}-$(/bin/cat _uuid) --format 'value(status.url)' \
+          --platform managed --region ${_SERVICE_REGION}
+    }
+    echo $(get_url) > _service_url
+    echo "Cloud Run URL for ${_SERVICE} is $(cat _service_url)"
+
+- id: 'Get Cloud Run External IP'
+  name: 'gcr.io/cloud-builders/kubectl'
+  entrypoint: '/bin/bash'
+  args:
+  - '-c'
+  - |
+    get_ip() {
+        gcloud container clusters get-credentials ${_CLUSTER} --zone ${_CLUSTER_LOCATION} --project ${PROJECT_ID} \
+        && kubectl get svc istio-ingress --namespace gke-system \
+          --output 'jsonpath={.status.loadBalancer.ingress[0].ip}'
+    }
+    echo $(get_ip) > _service_ip
+
+- id: 'Integration Tests'
+  # TODO: Update the following image name, entrypoint, and args to fit your testing needs
+  name: 'alpine:3'
+  entrypoint: '/bin/sh'
+  dir: '${_SAMPLE_DIR}'
+  args:
+  - '-c'
+  - |
+    echo "Add integration tests!"
+
+- id: 'Teardown'
+  name: 'gcr.io/cloud-builders/gcloud:$_CLOUDSDK_VERSION'
+  entrypoint: '/bin/bash'
+  args:
+  - '-c'
+  - |
+    set -x
+    gcloud --quiet container images delete gcr.io/${PROJECT_ID}/${_SERVICE}:$(/bin/cat _uuid)
+    gcloud --quiet run services delete ${_SERVICE}-$(/bin/cat _uuid) --platform gke
+    set +x
+    echo "View build details in the console: https://console.cloud.google.com/cloud-build/builds/${BUILD_ID}"
+
+# Uncomment if skipping teardown to associate build with container image.
+# images:
+# - 'gcr.io/${PROJECT_ID}/${_SERVICE}'
```

Comment:
Note: can't add the tag here

Reply:
What does that mean?

Reply:
The build will be associated with `gcr.io/${PROJECT_ID}/${_SERVICE}` rather than `gcr.io/${PROJECT_ID}/${_SERVICE}:UUID`. You can see this in the build info. I don't see any repercussions based on https://cloud.google.com/cloud-build/docs/build-config#images. The field is used to save images if not explicitly pushing to GCR or if the image is not produced fail the build.

Reply:
That's fair, see my note about how I'm thinking about the container image piece. Sending it momentarily.